crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -664,9 +664,25 @@ impl<T: TransactionOrdering> TxPool<T> {
         tx: &TransactionId,
     ) -> Option<Arc<ValidPoolTransaction<T::Transaction>>> {
         match pool {
-            SubPool::Queued => self.queued_pool.remove_transaction(tx),
+            SubPool::Queued => {
+                let tx = self.queued_pool.remove_transaction(tx);
+
+                // We trace here instead of in the `ParkedPool` struct directly, because the type is
+                // generic and it would not be possible to distinguish whether a transaction is
+                // being removed from the `BaseFee` pool, or the `Queued` pool.
+                trace!(target: "txpool", hash=?tx.as_ref().map(|tx| tx.transaction.hash()), "Removed transaction from the basefee pool");
```

Comment:
could we lift the trace and include the `pool` instead of adding the same trace to multiple arms?

Reply:
> could we lift the trace and include the pool instead of adding the same trace to multiple arms?

hmm yeah that makes sense